src/lib/rest/httpHeaderAdd.cpp

```diff
@@ -103,7 +105,12 @@ void httpHeaderLinkAdd(const char* _url)
     freeLinkP = true;
   }
 
-  sprintf(linkP, "<%s>; %s", url, LINK_REL_AND_TYPE);
+  if ((unsigned int)snprintf(linkP, urlBufLen, "<%s>; %s", url, LINK_REL_AND_TYPE) >= urlBufLen)
```

Comment:
I know we're not supposed to use sprintf, but, if you look at the logic before, you'll see that space enough is guaranteed before sprintf is called.

[ I have a stupid static analyzer that complains on snprintf every time sizeof is not used for the  second parameter, and it actually proposes that I use sizeof(linkP)  (which is 8 in a 64-bit env ... (sigh)) ]

About the "if condition", lines 109-113 seems like dead code. You can't reach that part, for two different reasons:
- The logic before assures the buffer is big enough (or we stop but to "out of memory"
- snprintf assures the response is never >= urlBufLen (as it's the 2nd param for snprintf)

Just do the snprintf - no checks needed (well, check for -1 if you went :))
Later we'll see what the static analyzer says (I'm about to remove it. It's a pain in the neck)

Reply:
Usually, gcc warnings are much better than many external static analyzers - the newer the compiler is, the better. 

The point here is not that the buffer size is calculated wrongly -> that's the other patch in this series (e293688). The point is that sprintf() has an return value of int, so it can not write more than INT_MAX bytes and you must make sure that the size of the things to be written is < INT_MAX.

Reply:
Well, INT_MAX ... that's a huge number (0x7FFFFFFF, I guess) - that would be 2 gigabytes.
Do you really think a comparison like that is necessary?
I mean, we'd have a SIGSEGV long before the snprintf finished copying chars if that happens :))
And apart from that, I think we can trust snprint to not overstep the limits we gave it in the second parameter.

Reply:
Well, my gcc 10.2 throws a warning and your -Werror makes it an error :-)

Reply:
In any case, do the check before calling s(n)printf.
After, it's just too late.
Not sure that will fix your warning though ...

Reply:
You can do the check afterwards as well, if need be (for the warning to go away).